core/src/plugins/filed/python/mariabackup/bareos-fd-mariabackup.py

```diff
@@ -237,32 +322,9 @@ def start_backup_job(self):
                     return bRC_Error
             # use old method as fallback, if module MySQLdb not available
             else:
-                get_lsn_command = (
-                    "echo 'SHOW ENGINE INNODB STATUS' | %s | grep 'Log sequence number' | awk '{ print $4 }'"
-                    % self.mysqlcmd
-                )
-                last_lsn_proc = Popen(
-                    get_lsn_command, shell=True, stdout=PIPE, stderr=PIPE
-                )
-                last_lsn_proc.wait()
-                returnCode = last_lsn_proc.poll()
-                (mysqlStdOut, mysqlStdErr) = last_lsn_proc.communicate()
-                if returnCode != 0 or mysqlStdErr:
-                    JobMessage(
-                        M_FATAL,
-                        'Could not get LSN with command "%s", Error: %s'
-                        % (get_lsn_command, mysqlStdErr),
-                    )
-                    return bRC_Error
-                else:
-                    try:
-                        last_lsn = int(mysqlStdOut)
-                    except:
-                        JobMessage(
-                            M_FATAL,
-                            'Error reading LSN: "%s" not an integer' % mysqlStdOut,
-                        )
-                        return bRC_Error
+                innodb_status = self.get_lsn_by_command()
+
+            last_lsn = self.parse_innodb_status(innodb_status)
```

Comment:
```suggestion
            last_lsn = self.parse_innodb_status(innodb_status)
            if last_lsn == bRC_Error:
                return bRC_Error
```
If an error occured, it should be returned here.